### PR TITLE
fix(Menu): horizontal keyboard navigation handling

### PR DESCRIPTION
**Pull request by johnsoter13**
# IMPORTANT

**Currently, this project is closed to any external contributions. Any pull request made against this project from external sources will likely be closed. If you would like to make changes to this project, please fork this project.**

# Guide

**This "Help" section can be deleted before submitting this pull request.**

*Update the name of this pull request to reflect the following shape:*

```
{type}/{scope?}/{message}
```

* **type** - A [conventional commit type](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum) **REQUIRED**
* **scope** - The kabab-case scope of the changes in this request
* **message** - A short, kebab-case statement describing the changes **REQUIRED**

*Provide a general summary of the scope of the changes in this pull request.*

* [Readme](https://github.com/momentum-design/momentum-react-v2/blob/master/README.md)
* [Getting Started Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/GETTING_STARTED.md)
* [Contributing Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/CONTRIBUTING.md)

# Description
In order to have consistent screen reader behavior. We determined that all single selection popouts should be menus, in order to do this we need to enable horizontal keyboard navigation for menus, similar to lists. 
*The full description of the changes made in this request.*
- Refactored horizontal keyboard handling into reusable hook
- added horizontal keyboard handling to menu/menu section
- added tests

# Links
[https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-549363](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-549363)
*Links to relevent resources.*

### CODE REVIEW

**Review by maxinteger — CHANGES_REQUESTED** [review]
Looks good, I have only 2 questions

File: src/components/Menu/Menu.types.ts [hunk 1]
```diff
@@ -42,6 +43,17 @@ export interface Props<T> extends AriaMenuProps<T> {
...
+   * Determines the orientation of the list
+   *
+   * The orientation of the list change the keyboard navigation in the list:
+   *
+   * - vertical: up and down arrow keys
+   * - horizontal: left and right arrow keys
+   *
+   * @default 'vertical'
+   */
+  orientation?: ListOrientation
```
[the remark this answers is not in the document]
Comment (johnsoter13) [reply src/components/Menu/Menu.types.ts:56]: I was asked to convert the In-meeting layout to a single selection menu. 

Screenshot 2024-08-14 at 10 36 49 AM

The top half of this menu is horizontal

File: src/hooks/useOrientationBasedKeyboardNavigation.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { setNextFocus } from '../components/List/List.utils';
+import { ListOrientation } from '../components/List/List.types';
+
+type IUseOrientationBasedKeyboardNavigationReturn = {
+  keyboardProps: HTMLAttributes<HTMLElement>,
+  getContext: () => {
+    listSize: number;
+    currentFocus: number;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    setContext: (newFocus: any) => void;
```
Comment (maxinteger) [inline src/hooks/useOrientationBasedKeyboardNavigation.ts:12]: Why did you used `any` for the `newFocus` type instead of `number`?

Comment (johnsoter13) [reply src/hooks/useOrientationBasedKeyboardNavigation.ts:12]: Ah I will change this. Pretty sure I get this from somewhere else in the code and didn't bother to change it

File: src/components/Menu/Menu.types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnsoter13) [reply src/components/Menu/Menu.types.ts:56]: I have it set to vertical as a default

**Review by maxinteger — APPROVED** [review]
Nice one!

**Pull request merged**